src/Raven.Server/Documents/Indexes/MapIndexBase.cs

```diff
@@ -139,6 +139,14 @@ private int UpdateIndexEntriesLucene(IndexItem indexItem, IEnumerable mapResults
                 numberOfOutputs++;
             }
 
+            if (indexItem.SkipLuceneDelete && numberOfOutputs > 0)
+            {
+                // we skip deleting from lucene for the initial indexing run
+                // but we must update the bloom filters if we actually indexed that document
+                using (_stats.BloomStats.Start())
+                    _filters.Add(indexItem.LowerId);
```

Comment:
I think that we can optimize this one as well :)
`Add` is checking if the id exists on all existing filters.
At the first indexing run, we know that it isn't possible. So we can just directly add it.
@ayende @ppekrol

Reply:
Check my other comment, it should solve your issue